src/main/java/com/synopsys/integration/alert/channel/jira/server/web/JiraServerCustomEndpoint.java

```diff
@@ -79,9 +81,17 @@ public JiraServerCustomEndpoint(JiraServerChannelKey jiraChannelKey, CustomEndpo
             PluginManagerService jiraAppService = jiraServicesFactory.createPluginManagerService();
             String username = jiraProperties.getUsername();
             String password = jiraProperties.getPassword();
-            Response response = jiraAppService.installMarketplaceServerApp(JiraConstants.JIRA_APP_KEY, username, password);
-            if (BooleanUtils.isTrue(response.isStatusCodeError())) {
-                return Optional.of(responseFactory.createBadRequestResponse("", "The Jira server responded with error code: " + response.getStatusCode()));
+            try {
+                Response response = jiraAppService.installMarketplaceServerApp(JiraConstants.JIRA_APP_KEY, username, password);
+                if (BooleanUtils.isTrue(response.isStatusCodeError())) {
```

Comment:
What determines if the status-code is an error or if an exception is thrown?

Reply:
This was from the existing code, it requires access to Response object which is now created on line 82 so we included it into the try-catch block.

Reply:
Right, but isn't 404 an error status-code? Why does an exception get thrown for a 404 and not for any other error code (like 400, 401, 500, etc). 

To clarify my questions: 
1. Will `jiraAppService.installMarketplaceServerApp(...)` only throw an exception for a 404 or will it also throw an exception for other errors?
2. Which errors does `response.isStatusCodeError()` work for if an exception is being thrown for some errors?

Reply:
From looking at the code the jira service will throw an exception for any response where the response code >= 400 a bad request.  The response.isStatusCodeError isn't needed here since if there is an error an exception will always be thrown.

Reply:
If that's the case, does that mean that before these changes, this never worked?

Reply:
It was likely there just in case the library changed.